aws-lc-rs/benches/kem_benchmark.rs

```diff
@@ -0,0 +1,150 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR ISC
+
+use aws_lc_rs::{test, test_file};
+use criterion::{criterion_group, criterion_main, Criterion};
+
+#[allow(non_camel_case_types)]
+#[derive(Copy, Clone, PartialEq, Eq, Debug)]
+enum Algorithm {
+    KYBER512_R3,
+}
+
+impl From<&str> for Algorithm {
+    fn from(value: &str) -> Self {
+        match value {
+            "KYBER512_R3" => Algorithm::KYBER512_R3,
+            _ => panic!("Unrecognized algorithm: '{value}'"),
+        }
+    }
+}
+
+pub struct KemConfig {
+    algorithm: Algorithm,
+    public_key: Box<[u8]>,
+    secret_key: Box<[u8]>,
+    ciphertext: Box<[u8]>,
```

Comment:
Is there a reason these can't be `Vec<u8>`? You are already cloning the `&[u8]` contents in the `new` constructor function by using `Vec::from`, so not sure why it then needs to be transformed into a box.

Reply:
Since we are never modifying the size, storing a `Vec` in the struct is unnecessary. A `Box` just contains a pointer to the data while a `Vec` contains a pointer, capacity, and size.